### ISSUE TITLE
How to read from a wxl file and use it in WixSharp

### ISSUE THREAD

**Issue by abskulkarni** [oleg-shilo/wixsharp#1486]
Hi @oleg-shilo 
I have below code for ManageProject.

![image](https://github.com/oleg-shilo/wixsharp/assets/23351437/396b9502-0f2f-47ff-ab30-cac24f714f87)

I want to have install dir path & source base directory to be dynamic coming from a wxl file. So,

1. How can I refer wxl file with dynamic path in C# ManagedProject code?
2. How to refer & use variables in wxl in C# code?

**Comment by abskulkarni** [comment]
Hi @oleg-shilo , can you please guide me regarding my query above?

**Comment by oleg-shilo** [comment]
You will need to implement the routine(s) that can extract your product name and source dir from the wxl file and then use these values to setup your build in the build script:

```C#
var productName = GetProductNameFromWxl(@"<path to your wxl>");
var sourceDir = GetSourceDirFromWxl(@"<path to your wxl>");

var project.= new ManagedProject("ManagedSetup",
    new Dir($@"%ProgramFiles%\Insurance Data Repository\{productName}"));

project..SourceBaseDir = sourceDir;

. . .
project.BuildMsi();
```

**Comment by abskulkarni** [comment]
Hi @oleg-shilo , do you mean to say I will have to write custom routine in my code to extract values from WXL? If yes, I will have to load WXL as normal XML in C# code and extract values using XPaths?

**Comment by oleg-shilo** [comment]
Yes, you will need to manipulate XML by yourself. Though I would prefer to do it via XDocument but any XML navigation routine is OK.
